File: kotlin-ide/j2k/new/testData/newJ2k/toKotlinClasses/iterableAndIterator2.java
//file
package demo;

import java.util.*;

class Test implements Iterable<String> {
    @Override
    public Iterator<String> iterator() {
        return null;
    }

    public Iterator<String> push(Iterator<String> i) {
        Iterator<String> j = i;
        return j;
    }
}

class FullTest implements java.lang.Iterable<String> {
    @Override
    public java.util.Iterator<String> iterator() {
        return null;
    }

    public java.util.Iterator<String> push(java.util.Iterator<String> i) {
        java.util.Iterator<String> j = i;
        return j;
    }
}

File: j2k/tests/testData/ast/toKotlinClasses/iterableAndIterator2.kt
package demo

import java.util.*
import kotlin.Iterator

class Test : Iterable<String> {
    override fun iterator(): Iterator<String> {
        return null
    }

    public fun push(i: Iterator<String>): Iterator<String> {
        val j = i
        return j
    }
}

class FullTest : Iterable<String> {
    override fun iterator(): Iterator<String> {
        return null
    }

    public fun push(i: Iterator<String>): Iterator<String> {
        val j = i
        return j
    }
}